packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -63,7 +66,7 @@ export interface UseGridColumnHeadersProps {
 }
 
 export interface GetHeadersParams {
-  position?: GridPinnedColumnPosition;
+  position?: PinnedColumnPosition;
   renderContext?: GridColumnsRenderContext;
   maxLastColumn?: number;
 }
```

Comment:
I'm not sure about this change, it modifies the public API. The old `GridPinnedColumnPosition` has only possible pinning values, where `PinnedColumnPosition` also includes internal implementation details such as `VIRTUAL` and `NONE`. It's annoying to have both types be named the same & represent the same thing, but I don't think we should make this change.

Reply:
And possibly our users can be checking with `=== 'left'` rather than `=== PinnedPosition.LEFT`. Thoughts @mui/xgrid ?

Reply:
Also it makes the constant part of the public API, so it would require the `Grid` prefix.

Reply:
Where does this leak to the public API? Must have not understood it, as it wasn't my intention to change the public types. I think public interface should be maximally constrained of course, but to an outside eye, it felt like there was so much translating happening between specific implementation of pinning and PinnedPosition currently, but for any internal implementation, it may not really worth the hassle imho.

Even if you would accidentally add a completely condition for a pinnedposition that can never exist a la `position === PinnedPosition.Whatever`, it doesn't have any impact, except for extraneous code. But the likelihood of doing that is probably next to none.

When I searched the docs, GridPinnedColumnPosition exists in two places, and those should remain unchanged, no?
https://mui.com/x/api/data-grid/grid-api/#grid-api-prop-isColumnPinned

`isColumnPinned` and `pinColumn`